### DISCUSSION TITLE
pmtiles Basemap for the whole  planet, with OSM id's, and OSM, or maptiler style, with all features

### DISCUSSION THREAD

**Discussion by amdevelopment-world** [protomaps/PMTiles#327] [Q&A]
Hi

i would like to create a pmtiles file for the whole world that would be used as a basemap
i want that basemap to look like the "outdoor" style of maptiler, like this:

https://www.maptiler.com/maps/#style=outdoor-v2&mode=2d&position=8.8/47.3961/7.6589

however, i need the OSM id's, i need to be able to know what the user is pointing at, and, i need all the data on OSM, every bench, statue, every path, all the details

i thought that would be the most basic usage of pm or mbtiles, and for OSM in general, but apparently it's not

i would like to use maplibre to display the map

to recap:

1) all features available on https://www.openstreetmap.org
2) i would like it to look, either like openstreetmap, or the outdoor style of maptiler
3) i need the OSM feature id

i tried planetiler, tippecanoe, without success, either the result is not a basemap, but a map that can be overlayed on a basemap
or the result is incomplete

frankly, i don't even know where to start

Martin

**Comment by bdon** [comment]
> i need all the data on OSM, every bench, statue, every path, all the details

That is not a design goal of most basemaps, since there is too much information in OSM to render efficiently.

**Reply by amdevelopment-world** [reply]
ok, i can compromise on the number of features, but i can't on style, and the feature ID

i would be happy with all ways, buildings, parks, and just art, no benches etc :)

the question is, how do i make a pmtiles basemap that displays well ? up to me to find out how many features i can put in there

**Reply by tmorell** [reply]
@bdon The source `protomaps` already has all features. With the proper filter I can see all POI that are in OSM.
When invoking `maplibre's queryRenderedFeatures` on the `pois` layer I get IDs that are not OSM. It is my understanding that that the rendering process creates it own IDs.
1. Is it possible to retrieve the OSM ID? Could it be attached as a property to the feature?
```json
{
    "type": "Feature",
    "id": 17594680218882,
    "geometry": { "type": "Point", "coordinates": [-4.1178855, 40.9476577] },
    "properties": {
        "osm_id": 2494174466,
        "osm_type": "node",
        "kind_detail": "regional",
        "kind": "restaurant",
        "min_zoom": 16,
        "name": "Mesón de Cándido"
    }
}
```

2. If not, are these IDs consistent between versions?

Thanks!

**Reply by wipfli** [reply]
The feature id is derived from the OpenStreetMap ID here:

https://github.com/protomaps/basemaps/blob/main/tiles/src/main/java/com/protomaps/basemap/feature/FeatureId.java

If you want to retrieve the OpenStreetMap Id from the Feature ID you can do something like this in JavaScript:

```
> console.log(BigInt(17594680218882) & ((1n << 44n) - 1n));
2494174466n
```

Questions around the protomaps basemap are best asked in the repository of the basemap, i.e., https://github.com/protomaps/basemaps/tree/main. Because this repo here is more about the `.pmtiles` format which can be used for any sort of tiles...

**Reply by tmorell** [reply]
@wipfli  This solves 100% the need that I have. And thanks for pointing me to the right repo.
